src/librustc_lint/types.rs

```diff
@@ -352,7 +376,16 @@ fn lint_literal<'a, 'tcx>(
             };
             if is_infinite == Ok(true) {
                 cx.struct_span_lint(OVERFLOWING_LITERALS, e.span, |lint| {
-                    lint.build(&format!("literal out of range for `{}`", t.name_str())).emit()
+                    lint.build(&format!("literal out of range for `{}`", t.name_str()))
+                        .note(&format!(
+                            "the literal `{}` does not fit into type `{}` as it is represents infinity",
```

Comment:
Is this a typo?

Reply:
I am not sure what is a typo. What I intended to do is adding more context for `f64` overflow.

Reply:
The "as it is represents infinity" part does not make sense to me. Does it mean the following?

```suggestion
                            "the literal `{}` does not fit into type `{}`. It will be converted to `{}::INFINITY`",
```

Reply:
Nice catch. Your error message is much more informative. Thanks. I will update it presently.